src/app/server/Server.cpp

```diff
@@ -515,11 +515,40 @@ void Server::ScheduleFactoryReset()
     PlatformMgr().ScheduleWork([](intptr_t) {
         // Delete all fabrics and emit Leave event.
         GetInstance().GetFabricTable().DeleteAllFabrics();
+        if (GetInstance().GetFabricTable().FabricCount() == 0)
+        {
+            ChipDeviceEvent event;
+            event.Type     = DeviceEventType::PublicEventTypes::kFactoryReset;
+            CHIP_ERROR err = PlatformMgr().PostEvent(&event);
+            if (err != CHIP_NO_ERROR)
+            {
+                ChipLogError(DeviceLayer, "Failed to post factory reset event");
+            }
+        }
         PlatformMgr().HandleServerShuttingDown();
         ConfigurationMgr().InitiateFactoryReset();
     });
 }
 
+void Server::ScheduleMatterDataReset()
```

Comment:
This has a lot in common with "ScheduleFactoryReset". Why is it different? The existing `Server::ScheduleFactoryReset()` is really the "initiate matter data reset" with a name that is not the best. The kMatterDataReset event should be triggered from the Server's FabricTable::Delegate, when last fabric removed.

It would then flow:

- ScheduleFactoryReset
- ScheduleWork
  - DeleteAllFabrics()
    - <That's it>
- ServerFabricDelegate::OnFabricRemoved()
  - Check if 0 fabrics left. If so:
    - Post event (kMatterDataReset)
    - ScheduleWork of:
        - PlatformMgr().HandleServerShuttingDown()
        - ConfigurationMgr().InitiateMatterDataReset()

The overall trigger of "full factory reset" should be done internally by the product, if, after everything is deleted, and `kMatterDataReset` event is seen, the *product* decides to factory reset.

In other words, we don't need a `Server::ScheduleFactoryReset` since "factory reset" is never triggered inside of Matter --> only "all matter data" can be reset, and whether this causes a factory reset is a choice of the product and not part of the spec.

Reply:
I agree to your point. Now, ScheduleFactoryReset will be replaced by ScheduleMatterReset and InitiateFactoryReset will be replaced by InitiateMatterReset
`ScheduleFactoryReset` ---------> `ScheduleMatterReset`
`InitiateFactoryReset`     ---------> `InitiateMatterReset`
which will erase matter data and `application will decide whether to erase additional data (Factory reset) or not.`

Since `matter reset` is mis-named to `factory reset`, some platforms deletes application data in matter reset. So with this change, is this expected that all the platforms need to update their matter reset logic?